### PR TITLE
Compiler support for decimal negate

### LINKED ISSUE #779: Implement unary minus for `decimal`
This requires adding a DecimalNegate insn.

We also need to add runtime support for this.

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
Fixes #779

### CODE REVIEW

File: compiler/modules/nback/number.bal [hunk 1]
```diff
@@ -12,6 +12,15 @@ final RuntimeFunction floatToIntFunction = {
     attrs: ["nounwind", "readnone", "speculatable", "willreturn"]
 };
 
+final RuntimeFunction decimalNegateFunction = {
+    name: "decimal_neg",
+    ty: {
+        returnType: LLVM_TAGGED_PTR,
+        paramTypes: [LLVM_TAGGED_PTR]
+    },
+    attrs: ["readnone"]
```
Comment (jclark) [inline compiler/modules/nback/number.bal]: readonly not readnone.  decimal_const is wrong too.

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/modules/nback/number.bal (hunk 1 above)
Comment (KavinduZoysa) [reply compiler/modules/nback/number.bal]: Changed

File: compiler/testSuite/11-decimal/neg1-v.bal [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import ballerina/io;
```
Comment (jclark) [inline compiler/testSuite/11-decimal/neg1-v.bal]: We are in subset 12 now, so 12-decimal

File: compiler/modules/nback/number.bal [hunk 3]
```diff
@@ -12,6 +12,15 @@ final RuntimeFunction floatToIntFunction = {
     attrs: ["nounwind", "readnone", "speculatable", "willreturn"]
 };
 
+final RuntimeFunction decimalNegateFunction = {
```
Comment (jclark) [inline compiler/modules/nback/number.bal]: This needs to be called decimalNegFunction (name is always same as runtime function name in camel case.)

File: compiler/testSuite/11-decimal/neg1-v.bal [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+public function main() {
```
Comment (jclark) [inline compiler/testSuite/11-decimal/neg1-v.bal]: You need to have cases where the operand is const.

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/testSuite/11-decimal/neg1-v.bal (hunk 2 above)
Comment (KavinduZoysa) [reply compiler/testSuite/11-decimal/neg1-v.bal]: Changed

File: compiler/modules/nback/number.bal (hunk 3 above)
Comment (KavinduZoysa) [reply compiler/modules/nback/number.bal]: Changed

File: compiler/testSuite/11-decimal/neg1-v.bal (hunk 4 above)
Comment (KavinduZoysa) [reply compiler/testSuite/11-decimal/neg1-v.bal]: Added test cases

**Pull request merged**